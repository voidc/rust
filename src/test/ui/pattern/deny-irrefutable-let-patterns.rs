#![deny(irrefutable_let_patterns)]

fn main() {
    if let _ = 5 {} //~ ERROR irrefutable `let` pattern

    while let _ = 5 {
        //~^ ERROR irrefutable `let` pattern
        break;
    }
}
