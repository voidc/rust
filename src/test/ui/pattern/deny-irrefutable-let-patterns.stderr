error: irrefutable `let` pattern
  --> $DIR/deny-irrefutable-let-patterns.rs:4:12
   |
LL |     if let _ = 5 {}
   |            ^
   |
note: the lint level is defined here
  --> $DIR/deny-irrefutable-let-patterns.rs:1:9
   |
LL | #![deny(irrefutable_let_patterns)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^

error: irrefutable `let` pattern
  --> $DIR/deny-irrefutable-let-patterns.rs:6:15
   |
LL |     while let _ = 5 {
   |               ^

error: aborting due to 2 previous errors

